bin/git-alias

```diff
@@ -1,16 +1,31 @@
 #!/usr/bin/env bash
 
+options=""
+
 usage() {
 cat <<HERE
-usage: git alias                         # list all aliases
-   or: git alias <search-pattern>        # show aliases matching pattern
-   or: git alias <alias-name> <command>  # alias a command
+usage: git alias [options]                          # list all aliases
+   or: git alias [options] <search-pattern>         # show aliases matching pattern
+   or: git alias [options] <alias-name> <command>   # alias a command
+options:
+    --global
+        For writing options: write to global ~/.gitconfig file
+        For reading options: read only from global ~/.gitconfig
+
+    --local
+        For writing options: write to the repository .git/config file
+        For reading options: read only from the repository .git/config
 HERE
 }
 
+if [[ "$1" == "--local" || "$1" == "--global" ]]; then
+    options=$1
+    shift
+fi
+
 case $# in
-  0) git config --get-regexp 'alias.*' | sed 's/^alias\.//' | sed 's/[ ]/ = /' | sort ;;
-  1) git alias | grep -e "$1" ;;
-  2) git config --global alias."$1" "$2" ;;
+  0) git config $options --get-regexp 'alias.*' | sed 's/^alias\.//' | sed 's/[ ]/ = /' | sort ;;
+  1) git config $options --get-regexp 'alias.*' | sed 's/^alias\.//' | sed 's/[ ]/ = /' | sort | grep -e "$1" ;;
+  2) git config $options alias."$1" "$2" ;;
```

Comment:
Same solution as above cause the `$parameter` is string already.

Reply:
The quote added here is just to make the shellcheck happy. Without it, the shellcheck will report `Double quote to prevent globbing and word splitting.`

Reply:
Understood, but in fact the `git config ""` will get the `error: key does not contain a section:  `. So I did not understand the happy shellcheck indeed.

Reply:
The `$option` won't contain any space, but shellcheck is not smart enough to figure out it. What we do is just to make it silent.